File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public class Hamming
{
    public static int Compute(string dna1, string dna2)
    {
        return dna1.ToCharArray().Zip(dna2.ToCharArray(), (c1, c2) => (c1 == c2) ? 0 : 1).Sum();
    }
}

File: fsharp/hamming/Hamming.fs
namespace Hamming

type Hamming() = 
    member x.compute(dna1, dna2) = 
        Seq.zip dna1 dna2 |>
        Seq.map (fun (gene1, gene2) -> if (gene1 = gene2) then 0 else 1) |>
        Seq.sum